### PR TITLE
feat(AvatarUpload): add size variant

### PR DESCRIPTION
**Pull request by ozgurkececioglu**
[FX-2889]

### Description

- `size` prop added with following values `small`, `large`
- disabled keyboard and drag functionality while `onEdit` functionality

### How to test

- go to [component page](https://picasso.toptal.net/fx-2889-implement-basic-avatar-upload/?path=/story/components-avatarupload--avatarupload)
- you can see the new story for sizes.
- HOVERING state of the component will be handled in upcoming PR

### Screenshots

Screen Shot 2022-08-29 at 12 34 09

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [x] Make sure that additions and changes on design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [x] Annotate all `props` in component with documentation
- [x] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- Covered with tests

**Breaking change**

- codemod is created and showcased in the changeset
- test alpha package of Picasso in StaffPortal

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-2889]: https://toptal-core.atlassian.net/browse/FX-2889?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/picasso/src/AvatarUpload/AvatarUpload.tsx [hunk 1]
```diff
@@ -157,21 +160,25 @@ export const AvatarUpload = forwardRef<HTMLDivElement, Props>(
       onDropAccepted: handleDropAccepted,
       onDropRejected: handleDropRejected,
       validator,
+      noClick: showAvatar && !showEditIcon,
+      // after showing avatar, only way to change the avatar is to use 'onEdit'
+      noDrag: showAvatar,
+      noKeyboard: showAvatar,
```
Comment (ozgurkececioglu) [inline packages/picasso/src/AvatarUpload/AvatarUpload.tsx:166]: This was a requirement but `onEdit` functionality story is still waiting to be added so I wanted to disable it before releasing the feature for the first time.

Comment (denieler) [reply packages/picasso/src/AvatarUpload/AvatarUpload.tsx:166]: better to have "positive" props 😃 `shouldClick`, `shouldDrag`, etc.

File: packages/picasso/src/AvatarUpload/DropzoneSvg/styles.ts [hunk 2]
```diff
@@ -4,15 +4,27 @@ export default ({ palette, transitions }: Theme) =>
...
+      margin: '-3px', // to center the svg
+    },
+    svgSmall: {
       width: '86px', // 6px for the outline stroke
       height: '86px', // 6px for the outline stroke
-      margin: '-3px', // to center the svg
+    },
+    svgLarge: {
+      width: '166px', // 6px for the outline stroke
+      height: '166px', // 6px for the outline stroke
```
Comment (denieler) [inline packages/picasso/src/AvatarUpload/DropzoneSvg/styles.ts:27]: can't we have size of svg `160px x 160px` as well, but maybe set viewBox to "0 0 166px, 166px" if necessary? I'm just worried about how much "actual" size our component would occupy in the layout. Wouldn't we have a situation, when 166px would be taken 🤔

**Comment by denieler** [comment]
I'm not sure whether it's correct, but to me a bit confusing, that my cursor is changed to `pointer` only when hovering this icon, but all the space around - still usual arrow cursor

Screenshot 2022-08-29 at 12 48 20

**Comment by ozgurkececioglu** [comment]
> I'm not sure whether it's correct but to me a bit confusing, that my cursor is changed to `pointer` only when hovering this icon, but all the space around - still the usual arrow cursor
> 
> Screenshot 2022-08-29 at 12 48 20
> 
> but I can click even around and it shows me the file upload dialog

This is because I split the hovering solution to upcoming PR and decided to not apply it in the first one to avoid double work 🤦‍♂️

File: packages/picasso/src/AvatarUpload/DropzoneSvg/styles.ts (hunk 2 above)
Comment (ozgurkececioglu) [reply packages/picasso/src/AvatarUpload/DropzoneSvg/styles.ts:27]: Hmm, good idea. I will give it a try 👍

**Comment by denieler** [comment]
> This is because I split the hovering solution to upcoming PR and decided to not apply it in the first one to avoid double work 🤦‍♂️

👍

File: packages/picasso/src/AvatarUpload/DropzoneSvg/styles.ts (hunk 2 above)
Comment (ozgurkececioglu) [reply packages/picasso/src/AvatarUpload/DropzoneSvg/styles.ts:27]: I did a quick check and prepared a test like this: 
Screen Shot 2022-08-29 at 12 59 04

I think it is not affecting the surrounding layout 👍

File: .changeset/blue-eels-compete.md [hunk 3]
```diff
@@ -7,3 +7,4 @@
 ### AvatarUpload
```
Comment (OndrejTuma) [inline .changeset/blue-eels-compete.md:7]: I dont see any packages specifies in this changeset 🙂 (and no line `---`)

File: packages/picasso/src/AvatarUpload/AvatarUpload.tsx (hunk 1 above)
Comment (ozgurkececioglu) [reply packages/picasso/src/AvatarUpload/AvatarUpload.tsx]: Can you check again @denieler? I didn't want to change them like:
```

    const shouldClick = !(showAvatar && !showEditIcon)

    ...

    noClick: !shouldClick
```

did you mean something else?

File: .changeset/blue-eels-compete.md (hunk 3 above)
Comment (OndrejTuma) [reply .changeset/blue-eels-compete.md:7]: You're right 🤦 sorry for the distraction

File: packages/picasso/src/AvatarUpload/AvatarUpload.tsx (hunk 1 above)
Comment (denieler) [reply packages/picasso/src/AvatarUpload/AvatarUpload.tsx]: I thought also that we need to change the prop name in the hook as well

Comment (ozgurkececioglu) [reply packages/picasso/src/AvatarUpload/AvatarUpload.tsx]: oh ok. but it is coming from 'react-dropzone' 🤔

**Comment by ozgurkececioglu** [comment]
@toptal-anvil ping reviewers

File: packages/picasso/src/AvatarUpload/AvatarUpload.tsx (hunk 1 above)
Comment (denieler) [reply packages/picasso/src/AvatarUpload/AvatarUpload.tsx]: they have bad API then 😁 ok, then we probably can't do much 🙇

**Review by OndrejTuma — APPROVED** [review]
Code looks good 🎉 checked temploy

**Review by dmaklygin — APPROVED** [review]

**Pull request merged**